src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -298,7 +298,7 @@ else if (msg.what == KILL_WORKER) {
                     ///////////////////////////
 
                     if ((returnCode >= mConfig.getBulkUploadLimit() || returnCode == MPDbAdapter.DB_OUT_OF_MEMORY_ERROR) &&
-                            SystemClock.elapsedRealtime() >= mRetryAfter) {
+                            SystemClock.elapsedRealtime() >= mRetryAfter && mFailedRetries <= 0) {
```

Comment:
Instead of implementing backoff by inserting a delayed FLUSH_QUEUE job, why not just implement backoff by the conceptual equivalent of

    mRetryAfter *= 2;

? (Implementation-wise, you'd keep track of the `mLastFlushTime` and ask whether `SystemClock.elapsedRealtime() >= mLastFlushTime + mRetryAfter`.)